### PR TITLE
Add reference documentation for global settings

### PR DESCRIPTION
**Pull request by charliermarsh**
## Summary

Second part of: https://github.com/astral-sh/uv/issues/5093.

### CODE REVIEW

File: crates/uv-settings/src/settings.rs [hunk 1]
```diff
@@ -56,12 +56,80 @@ pub struct Options {
 #[serde(rename_all = "kebab-case")]
 #[cfg_attr(feature = "schemars", derive(schemars::JsonSchema))]
 pub struct GlobalOptions {
+    /// Whether to load TLS certificates from the platform's native certificate store.
+    ///
+    /// By default, `uv` loads certificates from the bundled `webpki-roots` crate. The
```
Comment (zanieb) [inline crates/uv-settings/src/settings.rs:61]: I thought we were explicitly not using backticks for uv anywhere.

File: crates/uv-settings/src/settings.rs [hunk 2]
```diff
@@ -56,12 +56,80 @@ pub struct Options {
...
+    /// Disable network access, relying only on locally cached data and locally available files.
+    #[option(
+        default = "false",
+        value_type = "bool",
+        example = r#"
+            offline = true
+        "#
+    )]
     pub offline: Option<bool>,
+    /// Avoid reading from or writing to the cache.
```
Comment (zanieb) [inline crates/uv-settings/src/settings.rs:85]: Should we note that we use a temporary cache for in-process purposes?

File: crates/uv-settings/src/settings.rs (hunk 1 above)
Comment (charliermarsh) [reply crates/uv-settings/src/settings.rs:61]: Haha. This is copied from the CLI... I'll have to change it there too.

**Comment by charliermarsh** [comment]
Changed existing backticked uv references, removing the backticks.

**Pull request merged**